spec.html

```diff
@@ -13954,6 +13957,57 @@ <h1>
     </emu-clause>
   </emu-clause>
 
+  <emu-clause id="sec-built-in-async-function-objects">
+    <h1>Built-in Async Function Objects</h1>
+    <p><dfn variants="built-in async function object,built-in async function objects">Built-in async function objects</dfn> are built-in function objects that provide alternative [[Call]] and [[Construct]] internal methods that conform to the following definitions:</p>
+
+    <emu-clause id="sec-built-in-async-function-objects-call" type="internal method">
+      <h1>
+        [[Call]] (
+          _thisArgument_: an ECMAScript language value,
+          _argumentsList_: a List of ECMAScript language values,
+        ): a normal completion containing an ECMAScript language value
```

Comment:
What do you think about this?

```suggestion
        ): a normal completion containing a Promise
```

Reply:
I personally like it, but I can see how some people would prefer all overloads of concrete methods to have the same signature.

Reply:
Environment Records have lots of examples of concrete methods with different (but compatible) signatures.

E.g., the five implementations of `HasThisBinding` have 3 different return types: `*false*`, `*true*`, and `a Boolean`.

Reply:
"a Promise" SGTM